src/Integration.UnitTests/LocalServices/RuleSetInspectorTests.cs

```diff
@@ -0,0 +1,494 @@
+
+            // Verify
+            AssertNoConflicts(conflicts);
+        }
+
+        [TestMethod]
+        public void RuleSetInspector_FindConflictingRules_ComplexStructureWithConflicts()
+        {
+            // Setup
+            string solutionRuleSet = this.CreateCommonRuleSet().FilePath;
+            this.temporaryFiles.AddFile(solutionRuleSet, false);
+
+            // Modifies all the solution rules to Info (should not impact the result)
+            RuleSet otherRuleSet = this.CreateCommonRuleSet("User.ruleset", RuleAction.Info);
+            this.temporaryFiles.AddFile(otherRuleSet.FilePath, false);
+
+            // Modifies all the solution rules to None (should not impact the result)
+            const string BuiltInRuleSetName = "NoneAllRules.ruleset";
+            RuleSet vsRuleSet = CreateVsRuleSet(DefaultNumberOfRules, BuiltInRuleSetName, RuleAction.None);
+            this.temporaryFiles.AddFile(vsRuleSet.FilePath, false);
+
+            // The project has 3 modification -> error, info and warning
+            RuleSet projectRuleSet = CreateProjectRuleSetWithIncludes(DefaultNumberOfRules, solutionRuleSet, IncludeType.AsIs, RuleAction.Error, otherRuleSet.FilePath);
+            projectRuleSet.Rules[1].Action = RuleAction.Info;
+            projectRuleSet.Rules[2].Action = RuleAction.None;
+            projectRuleSet.WriteToFile(projectRuleSet.FilePath);
+            this.temporaryFiles.AddFile(projectRuleSet.FilePath, false);
+
+            // Act 
+            RuleConflictInfo conflicts = this.testSubject.FindConflictingRules(solutionRuleSet, projectRuleSet.FilePath);
+
+            // Verify [verify that having all that with all that extra noise, since the project level overrides two of the values there will be conflicts]
+            AssertWeakRulesByFullIds(conflicts, new[] { projectRuleSet.Rules[1] });
+            AssertMissingRulesByFullIds(conflicts, new[] { projectRuleSet.Rules[2] });
+        }
+
+        [TestMethod]
+        public void RuleSetInspector_FindConflictingRules_RuleSetFileCustomization_BaselineRuleSetWasRemoved()
+        {
+            // Setup
+            RuleSet solutionRuleSet = this.CreateCommonRuleSet();
+            this.temporaryFiles.AddFile(solutionRuleSet.FilePath, false);
+
+            RuleSet otherRuleSet = this.CreateCommonRuleSet($"User.ruleset", rules: 2);
+            this.temporaryFiles.AddFile(otherRuleSet.FilePath, false);
+
+            RuleSet projectRuleSet = CreateProjectRuleSetWithIncludes(0, otherRuleSet.FilePath, IncludeType.AsIs);
+            this.temporaryFiles.AddFile(projectRuleSet.FilePath, false);
+
+            // Act
+            RuleConflictInfo conflicts = this.testSubject.FindConflictingRules(solutionRuleSet.FilePath, projectRuleSet.FilePath);
+
+            // Verify [deleting the baseline and not providing any replacements means that the delta rules are missing]
+            AssertMissingRulesByFullIds(conflicts, new[] { solutionRuleSet.Rules.Last() });
```

Comment:
It would be clearer why the last solution rule is missing if the solution ruleset was created by passing in 3 rather than relying on that being the default number of rules (as well as being more robust if the default number of rules is ever changed).